### PR TITLE
feat(tests): Adds more Jest tests and restructures API config import for checkout

### PR DESCRIPTION
**Pull request by cdarland**

### CODE REVIEW

File: packages/checkout/src/components/__test_helpers__/helpers.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import { render } from "@testing-library/react";
+
+const renderWithRedux = ( ui, { initialState, store = createStore(reducer, initialState) } = {} ) => {
+  return {
+    ...render(<Provider store={store}>{ui}</Provider>),
+    store,
+  }
+};
+
+export { renderWithRedux };
```
Comment (jacksonhardaker) [inline packages/checkout/src/components/__test_helpers__/helpers.js]: I think it probably makes more sense if this were in a root level `./__mocks__` folder. What do you think? We're probably likely to use this in a few places?

File: packages/checkout/src/components/app.js [hunk 2]
```diff
@@ -77,19 +77,20 @@ class App extends Component {
...
+    cart: PropTypes.array,
+    config: PropTypes.object
   }
 
   componentDidMount() {
     this.props.dispatch(loadCustomer());
     cartFromParams();
     this.props
-      .dispatch(loadProducts())
+      .dispatch(loadProducts(this.props.config))
```
Comment (jacksonhardaker) [inline packages/checkout/src/components/app.js]: Should be the new method of fetching config?

File: packages/minicart/src/components/__test_helpers__/helpers.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import { render } from "@testing-library/react";
+
+const renderWithRedux = ( ui, { initialState, store = createStore(reducer, initialState) } = {} ) => {
+  return {
+    ...render(<Provider store={store}>{ui}</Provider>),
+    store,
+  }
+};
+
+export { renderWithRedux };
```
Comment (jacksonhardaker) [inline packages/minicart/src/components/__test_helpers__/helpers.js]: Looks like we can go even further and make this live in the highest level root `./__mocks__`?

File: packages/checkout/src/components/__test_helpers__/helpers.js (hunk 1 above)
Comment (cdarland) [reply packages/checkout/src/components/__test_helpers__/helpers.js]: Good call. We will likely use that all over the place

**Comment by cdarland** [comment]
Jest testing structure reviewed and approved by Aimee

**Pull request merged**